lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,182 @@
+$type = Get-AnsibleParam $params "type" -type "str" -Default "element" -ValidateSet "element", "attribute", "text"
+$attribute = Get-AnsibleParam $params "attribute" -type "str" -FailIfEmpty ($type -eq "attribute")
+
+$result = @{
+    changed = $false
+}
+
+If (-Not (Test-Path -Path $dest -PathType Leaf)){
+    Fail-Json $result "Specified path $dest does not exist or is not a file."
+}
+
+[xml]$xmlorig = Get-Content -Path $dest
+$xmlnew = $xmlorig.Clone()
+if ($type -eq "element") {
+    $xmlchild = [xml]$fragment
+
+    $child = $xmlnew.CreateElement($xmlchild.DocumentElement.Name, $xmlnew.DocumentElement.NamespaceURI)
+    foreach ($attr in $xmlchild.DocumentElement.Attributes) {
+        $child.SetAttribute($attr);
+    }
+
+    foreach ($node in $xmlchild.DocumentElement.ChildNodes) {
+        $newnode = $xmlnew.CreateElement($node.Name, $xmlnew.DocumentElement.NamespaceURI)
+        foreach ($attr in $node.Attributes) {
+            $newnode.SetAttribute($attr)
+        }
+        $newnode.InnerText = $node.InnerText
+        $child.AppendChild($newnode) | Out-Null
+    }
+
+    $node = $xmlnew.SelectSingleNode($xpath)
+    $elements = $node.ChildNodes
+    [bool]$add = $TRUE
+    if ($elements.Count) {
+        foreach ($element in $elements) {
+            try {
+                Compare-XmlDocs $xmlchild.DocumentElement $element
+                $add = $FALSE
+                break
+            } catch {
+                $_.InvocationInfo | Out-File c:\win_xml.log
+            }
+        }
+        if ($add) {
+            [void]$node.AppendChild($child)
+        }
+    } else {
+        [void]$node.AppendChild($child)
+    }
+
+    if ($add) {
+        $result.changed = $TRUE
+        if (-Not $check_mode) {
+            if ($backup -eq "yes") {
+                $result.backup = BackupFile($dest)
+            }
+            $result.msg = "xml added"
+            $xmlnew.Save($dest)
+        } else {
+            $result.msg = "added check mode"
+        }
+    } else {
+        $result.msg = "already present"
+    }
+} elseif ($type -eq "text") {
+    $node = $xmlnew.SelectSingleNode($xpath)
+    [bool]$add = ($node.InnerText -ne $fragment)
+    if ($add) {
+        $result.changed = $TRUE
+        if (-Not $check_mode) {
+            if ($backup -eq "yes") {
+                $result.backup = BackupFile($dest)
+            }
+            $node.InnerText = $fragment
+            $xmlnew.Save("$dest")
+            $result.msg = "text changed"
+        } else {
+            $result.msg = "text changed check mode"
+        }
+    } else {
+        $result.msg = "not changed"
+    }
+} elseif ($type -eq "attribute") {
+    $node = $xmlnew.SelectSingleNode($xpath)
+    [bool]$add = !$node.HasAttribute($attribute) -Or ($node.$attribute -ne $fragment)
+    if ($add) {
+        $result.changed = $TRUE
+        if (-Not $check_mode) {
+            if ($backup -eq "yes") {
```

Comment:
Now that `$backup` is a boolean, this should be `if ($backup -eq $true)` or just if `if ($backup)`